### PR TITLE
Changing default plot style to double logarithmic

### LINKED ISSUE #124: X values for logarithmic plots
When user selects logaritmic scale on horizontal scale, the series of energies (`xs`) for which stopping power is generated should be a geometric series, not arithmetic one.

### LINKED ISSUE #128: Change default plotting style to double logarithmic

### PR DESCRIPTION
**Pull request by Mexolius**
closes #128
closes #124

### CODE REVIEW

File: src/Backend/DataSeriesFactory.js [hunk 1]
```diff
@@ -2,7 +2,19 @@ export default class TraceFactory {
     static getXValuesByPoints(minValue, maxValue, numberOfPoints) {
         if(numberOfPoints === 1) return [(maxValue + minValue)/2.0]
         const interval = (maxValue - minValue) / (numberOfPoints - 1)
-        return Array.from(new Array(~~numberOfPoints), (_, key) => { return minValue + interval * key })
+        return Array.from(new Array(numberOfPoints), (_, key) => { return minValue + interval * key })
+    }
+
+    static getLogXValuesByPoints(minValue, maxValue, numberOfPoints){
+        const values = new Array(numberOfPoints);
```
Comment (grzanka) [inline src/Backend/DataSeriesFactory.js:9]: if number of points is 1, we should return geometric mean of minValue and maxValue, namely sqrt( minValue * maxValue)

**Comment by grzanka** [comment]
I selected 5 points, but only 3 are visible:

image

**Comment by grzanka** [comment]
First I set 100 points
image

the I change the X scale to linear:
image

I would expect that the points on X axis would be recrated as arithmetic series and stopping power would be recalculated again

File: src/Components/Pages/StoppingPower.js [hunk 2]
```diff
@@ -55,21 +66,35 @@ class StoppingPowerComponent extends React.PureComponent {
         return { xAxis, yAxis, method }
     }
 
+    async onXAxisChange(xAxis){
+        const dataSeries = await Promise.all(this.state.dataSeries.map(async ({data,metadata})=>{
+            const {color, isShown, index, name} = data
+            const newData = Object.assign(
+                {color, isShown, index, name}
+                , await this.wrapper.getDataSeries(metadata, xAxis===1)
```
Comment (grzanka) [inline src/Components/Pages/StoppingPower.js:74]: It is hard to guess what `xAxis===1` means. Is it log scale or linear scale. `1` is a bit meaningless.

**Comment by grzanka** [comment]
Can you fix CMake problem:
```gitpod /workspace/dedx_web $ {
> cd $HOME/emsdk && . ./emsdk_env.sh && cd /workspace/dedx_web/ && ./build_wasm.sh && gp sync-done wasm
> 
> }; exit
Adding directories to PATH:
PATH += /home/gitpod/emsdk
PATH += /home/gitpod/emsdk/upstream/emscripten
PATH += /home/gitpod/emsdk/node/14.15.5_64bit/bin

Setting environment variables:
PATH = /home/gitpod/emsdk:/home/gitpod/emsdk/upstream/emscripten:/home/gitpod/emsdk/node/14.15.5_64bit/bin:/home/gitpod/.pyenv/versions/3.8.12/bin:/home/gitpod/.pyenv/libexec:/home/gitpod/.pyenv/plugins/python-build/bin:/home/gitpod/.pyenv/plugins/pyenv-virtualenv/bin:/home/gitpod/.pyenv/plugins/pyenv-update/bin:/home/gitpod/.pyenv/plugins/pyenv-installer/bin:/home/gitpod/.pyenv/plugins/pyenv-doctor/bin:/home/gitpod/.sdkman/candidates/maven/current/bin:/home/gitpod/.sdkman/candidates/java/current/bin:/home/gitpod/.sdkman/candidates/gradle/current/bin:/home/gitpod/.rvm/gems/ruby-2.7.4/bin:/home/gitpod/.rvm/gems/ruby-2.7.4@global/bin:/home/gitpod/.rvm/rubies/ruby-2.7.4/bin:/home/gitpod/.pyenv/plugins/pyenv-virtualenv/shims:/ide/bin:/workspace/.cargo/bin:/workspace/.rvm/bin:/workspace/.pip-modules/bin:/home/gitpod/.pyenv/bin:/home/gitpod/.pyenv/shims:/workspace/go/bin:/home/gitpod/go/bin:/home/gitpod/go-packages/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin/:/home/gitpod/.nvm/versions/node/v16.13.0/bin:/home/gitpod/.cargo/bin:/usr/games:/home/gitpod/.rvm/bin
EMSDK = /home/gitpod/emsdk
EM_CONFIG = /home/gitpod/emsdk/.emscripten
EMSDK_NODE = /home/gitpod/emsdk/node/14.15.5_64bit/bin/node
configure: cmake . -B ./build -DCMAKE_TOOLCHAIN_FILE=/home/gitpod/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/home/gitpod/emsdk/node/14.15.5_64bit/bin/node
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found Subversion: /usr/bin/svn (found version "1.13.0") 
CMake Deprecation Warning at python/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS) 
-- Python libraries NOT found
-- Found PythonInterp: /home/gitpod/.pyenv/versions/3.8.12/bin/python (found version "3.8.12") 
CMake Error at /home/linuxbrew/.linuxbrew/Cellar/cmake/3.21.4/share/cmake/Modules/CPack.cmake:635 (message):
  CPack package description file: "/workspace/dedx_web/libdedx/README" could
  not be found.
Call Stack (most recent call first):
  /home/linuxbrew/.linuxbrew/Cellar/cmake/3.21.4/share/cmake/Modules/CPack.cmake:639 (cpack_check_file_exists)
  CMakeLists.txt:122 (INCLUDE)


-- Configuring incomplete, errors occurred!
See also "/workspace/dedx_web/libdedx/build/CMakeFiles/CMakeOutput.log".
emcmake: error: 'cmake . -B ./build -DCMAKE_TOOLCHAIN_FILE=/home/gitpod/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/home/gitpod/emsdk/node/14.15.5_64bit/bin/node' failed (returned 1)
make: cmake --build .
make: Makefile: No such file or directory
make: *** No rule to make target 'Makefile'.  Stop.
emmake: error: 'cmake --build .' failed (returned 2)
shared:INFO: (Emscripten: Running sanity checks)
emcc: error: libdedx.a: No such file or directory ("libdedx.a" was expected to be an input file, based on the commandline arguments provided)
total 0
drwxr-xr-x 3 gitpod gitpod  24 Dec  8 15:39 .
drwxr-xr-x 6 gitpod gitpod 177 Dec  8 15:39 ..
drwxr-xr-x 2 gitpod gitpod   6 Dec  8 15:39 CMakeFiles
cp: cannot stat './libdedx/build/libdedx/weblibdedx.js': No such file or directory
cp: cannot stat './libdedx/build/libdedx/weblibdedx.wasm': No such file or directory
sed: can't read ./src/Backend/weblibdedx.js: No such file or directory
sed: can't read ./src/Backend/weblibdedx.js: No such file or directory
sed: can't read ./src/Backend/weblibdedx.js: No such file or directory
exit

🤙 This task ran as a workspace prebuild
```

File: .github/workflows/test_and_deploy.yml [hunk 3]
```diff
@@ -109,9 +123,17 @@ jobs:
 
       - uses: actions/download-artifact@v2
         with:
-          name: wasm bin
+          name: wasm public
+          path: public
+
+<<<<<<< HEAD
```
Comment (grzanka) [inline .github/workflows/test_and_deploy.yml:129]: Is that leftover of some merge ?

Comment (Mexolius) [reply .github/workflows/test_and_deploy.yml]: Yeah, those are some weird commits from way behind - must have been on the local branch and i didn't notice them. I hope nothing else is broken

File: src/Components/Pages/StoppingPower.js (hunk 2 above)
Comment (Mexolius) [reply src/Components/Pages/StoppingPower.js]: Done

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**